### DISCUSSION TITLE
Cannot find module '@babel/helper-environment-visitor'

### DISCUSSION THREAD

**Discussion by christianinsert** [babel/babel#14069] [Q&A] [answered]
Recently I see in my CI environment the error below on all PRs on that repo:

```
[BABEL] .../node_modules/react-scripts/node_modules/react-app-polyfill/jsdom.js: Cannot find module '@babel/helper-environment-visitor'
    Require stack:
    - .../node_modules/@babel/core/node_modules/@babel/traverse/lib/path/conversion.js
    - .../node_modules/@babel/core/node_modules/@babel/traverse/lib/path/index.js
    - .../node_modules/@babel/core/node_modules/@babel/traverse/lib/context.js
    - .../node_modules/@babel/core/node_modules/@babel/traverse/lib/index.js
    - .../node_modules/@babel/core/lib/index.js
    - .../node_modules/@babel/plugin-proposal-async-generator-functions/lib/index.js
    - .../node_modules/babel-preset-react-app/node_modules/@babel/preset-env/lib/available-plugins.js
    - .../node_modules/babel-preset-react-app/node_modules/@babel/preset-env/lib/plugins-compat-data.js
    - .../node_modules/babel-preset-react-app/node_modules/@babel/preset-env/lib/normalize-options.js
    - .../node_modules/babel-preset-react-app/node_modules/@babel/preset-env/lib/index.js
    - .../node_modules/babel-preset-react-app/create.js
    - .../node_modules/babel-preset-react-app/index.js
    - .../node_modules/react-scripts/node_modules/@babel/core/lib/config/files/plugins.js
    - .../node_modules/react-scripts/node_modules/@babel/core/lib/config/files/index.js
    - .../node_modules/react-scripts/node_modules/@babel/core/lib/index.js
    - .../node_modules/react-scripts/node_modules/babel-jest/build/loadBabelConfig.js
    - .../node_modules/react-scripts/node_modules/babel-jest/build/index.js
    - .../node_modules/react-scripts/config/jest/babelTransform.js
    - .../node_modules/@jest/transform/build/ScriptTransformer.js
    - .../node_modules/@jest/transform/build/index.js
    - .../node_modules/jest-runtime/build/index.js
    - .../node_modules/@jest/core/build/cli/index.js
    - .../node_modules/@jest/core/build/jest.js
    - .../node_modules/jest-cli/build/cli/index.js
    - .../node_modules/jest-cli/build/index.js
    - .../node_modules/jest/build/jest.js
    - .../node_modules/react-scripts/scripts/test.js (While processing: ".../node_modules/babel-preset-react-app/index.js")
 ```

The same command which fails on the CI works locally for me, though. And the strange thing is: I cannot find a module named `@babel/helper-environment-visitor` anywhere: neither in `yarn.lock` nor in the source code of `nodue_modules` nor in the globally installed packages. And all Google hits are suspicious (to say the least).

So, what is going on here? Any help appreciated.

**Comment by nicolo-ribaudo** [comment, marked as answer]
In an old `@babel/traverse` version we forgot to mark that package as a dependency. I suspect for some reason (an old cache?) your CI is picking an old `@babel/traverse` version. I suggest deleting it from your lockfile, running `yarn install` (so that it picks up the new version) and see if it solves the problem.

**Reply by christianinsert** [reply]
Thanks for your help. Strange enough, updating react-scripts actually helped.

**Reply by vala** [reply]
Thanks for the tip, in my case I had to update another babel package, which was the one mentionned in the last (bottom) line of the require stack in the `(While processing ...)` portion of the line.

**Reply by k3n** [reply]
Any idea which version of `@babel/traverse` that was?

**Reply by nicolo-ribaudo** [reply]
I think we fixed it in v7.16.5

**Reply by k3n** [reply]
So all versions prior to `v7.16.5` are affected? Original comment made it sound like a one-off version had the defect. Thank you.

**Comment by evilPaprika** [comment]
`rm -rf node_modules`
`yarn install`
\- helped in my case

**Reply by Drew-Daniels** [reply]
This alone solved it for me.

**Reply by liang7hi** [reply]
This also works for me.

**Comment by mismailswe** [comment]
First I removed babel/traverse from pakage.lock and then delete the node-module and install it again. This worked for me
